server/app/repository/AccountRepository.java

```diff
@@ -246,6 +252,45 @@ public List<TrustedIntermediaryGroupModel> listTrustedIntermediaryGroups() {
         .findList();
   }
 
+  public PaginationResult<TrustedIntermediaryGroupModel> getAllTiGroupsWithinPageSpec(
+      F.Either<IdentifierBasedPaginationSpec<Long>, PageNumberBasedPaginationSpec>
+          paginationSpecEither) {
+
+    var query =
+        database
+            .find(TrustedIntermediaryGroupModel.class)
+            .setLabel("TrustedIntermediaryGroupByPage.findListByPage")
+            .orderBy("name asc")
+            .setProfileLocation(
+                queryProfileLocationBuilder.create("listTrustedIntermediaryGroupsByPage"));
+
+    PagedList<TrustedIntermediaryGroupModel> pagedQuery;
+
+    if (paginationSpecEither.left.isPresent()) {
+      IdentifierBasedPaginationSpec<Long> paginationSpec = paginationSpecEither.left.get();
+      pagedQuery =
+          query
+              .where()
+              .lt("id", paginationSpec.getCurrentPageOffsetIdentifier())
+              .setMaxRows(paginationSpec.getPageSize())
+              .findPagedList();
+    } else {
```

Comment:
Question: Is it possible that both left and right are empty? If so, should we check that case?